python_scripts/shellies_discovery.py

```diff
@@ -1075,11 +1075,11 @@
             '"avty_t":"' + availability_topic + '",'
             '"pl_avail":"true",'
             '"pl_not_avail":"false",'
-            '"cmd_on_tpl":"{\\"turn\\":\\"on\\"{% if brightness is defined %},\\"brightness\\":{{brightness|float|multiply(0.3922)|round}}{% endif %}{% if white_value is defined %},\\"white\\":{{white_value|float|multiply(0.3922)|round}}{% endif %}}",'
+            '"cmd_on_tpl":"{\\"turn\\":\\"on\\"{% if brightness is defined %},\\"brightness\\":{{brightness|float|multiply(0.3922)|round}}{% endif %}{% if color_temp is defined %},\\"temp\\":{{[([((1000000/(color_temp|int))|round(0,\\"floor\\")),6500]|min),2700]|max}}{% endif %}}",'
             '"cmd_off_tpl":"{\\"turn\\":\\"off\\"}",'
             '"stat_tpl":"{% if value_json.ison %}on{% else %}off{% endif %}",'
             '"bri_tpl":"{{value_json.brightness|float|multiply(2.55)|round}}",'
-            '"white_value_template":"{{value_json.white|float|multiply(2.55)|round}}",'
+            '"color_temp_template":"{{[([((1000000/(value_json.temp|int))|round(0,\\"floor\\")),370]|min),153]|max}}",'
```

Comment:
It's not necessary to check that the resulting value is between 153 and 370.

Reply:
You are right. It is only important other way round. On the other hand, when HA will start supporting mired boundaries you will have boundaries in code so you can easily move them into proper properties without re-inventing them ;)